src/OrchardCore.Modules/OrchardCore.Forms/Services/IValidationRuleProvider.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using OrchardCore.Forms.Services.Models;
+
+namespace OrchardCore.Forms.Services
+{
+    public interface IValidationRuleProvider
```

Comment:
It looks like it should use the Options pattern, and configuration (startup) would register these options. That would make localized strings easier to define as you can inject the T in a Startup class, and also prevent DI usage.